### PR TITLE
docs(store): update hmr guide

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

https://github.com/ngxs/store/issues/567

## What is the new behavior?

state saving when reloading HMR

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by eranshmil** [comment]
A few days ago I did a lot of testing to try to solve this issue, and it worked most of the time.
The main issue is lazy-loaded states that were loaded before the hmr reloaded the code, and are lazy loaded afterwards and raises the @@UPDATE_STATE action. Then they were emptied from the previous state.

Hope I was clear enough.

**Comment by splincode** [comment]
https://stackblitz.com/edit/angular-hmr-wth-ngxs?file=src%2Fmain.ts

**Comment by marcus-sa** [comment]
Why would one use `.filter` over `.find` in this situation?

File: docs/advanced/hmr.md [hunk 1]
```diff
@@ -99,23 +99,87 @@ Install the @angularclass/hmr module as a dev-dependency
...
     makeVisible();
   });
 };
+
+function hmrNgxsStore(mod: NgModuleRef<any>, outlet: StoreHrmOutlet): void {
+  let storeRef: AppStoreRef | null = null;
+
+  const moduleRef: any = mod as ModuleRef;
+  const providers = moduleRef._providers
+    .filter((provider: any) => provider instanceof Object && provider.constructor.name === 'Store');
```
Comment (marcus-sa) [inline docs/advanced/hmr.md]: Why would you use `.filter` over `.find`?

**Pull request closed without merging**